### PR TITLE
new "B97-V" based D3BJ functionals 

### LINKED ISSUE #1188: user supplied dict. functionals don't work with custom DFT-NL
```
molecule custom {
0 1
   AR         0.00000000      0.00000000      0.00000000
}

set basis def2-SVP
set DFT_VV10_B 4.0
pbe0 = {
    "name": "my_PBE0",
    "x_functionals": {"GGA_X_PBE": {"alpha": 0.75}},
    "x_hf": {"alpha": 0.25},
    "c_functionals": {"GGA_C_PBE": {}}
}
func_call = energy('SCF', dft_functional=pbe0)
```

Logic check for `isinstance(name, dict)` is missing, i think? I wasn't fully aware of how it works.
https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/dft_funcs/superfuncs.py#L113

### PR DESCRIPTION
**Pull request by hokru**
## Description
Adds Goerigk's D3 versions of B97M-V, wB97M-V, wB97X-V
https://pubs.acs.org/doi/10.1021/acs.jctc.8b00842.

Absolute and relative energies added from L. Georigk. S22 MADs in SI are also reproduced.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] fixes #1188 
- [x] B97M-D3(BJ), wB97M-D3(BJ)
- [x] wB97X-D3(BJ) (of wB97X-V origin)
- [x] revTPSS-NL parameters from Kruse,Banas,Sponer (JCTC 'just accepted')
- [x] moves rest of the post-scf NL logic to python.

## Checklist
- [x] Tests added for any new features
- [x]  [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by hokru** [comment]
The dispersion refactoring apparently changes the way I need to turn of VV10 from libxc. Will need to track down the changes to make it work again.

**Comment by hokru** [comment]
turning off VV10 in the functional and having `DFT_VV10_POSTSCF` active will crash and not yet sure how to catch this elegantly.

**Comment by dgasmith** [comment]
Do you have a reproducible example?

**Comment by hokru** [comment]
any non-VV10 DFT functional currently (release v1.2.1) crashes already if one sets `DFT_VV10_POSTSCF true`, simply because VV10 is turned on without any checks. 

And while writing about the problem with the modified libxc functionals, I figured i need to switch two lines of code!
What got me stuck, was that using `set_vv10_b` in `dict_builder.py` re-activated VV10, and I need to use it before turning off VV10.

Now waiting on the reference outputs from Lars to make a test.

**Comment by dgasmith** [comment]
Cool, glad everything worked out! Is setting `coeffs` turning on vv10 opaque? It originally seemed like a good idea, but not if it is causing issues.

**Comment by hokru** [comment]
> Is setting coeffs turning on vv10 opaque? 

It's useful i think, I even used it for my initial DFT-NL. Just forgot about it.

**Comment by hokru** [comment]
This is ready for review. Mind the updated OP.

I want to wait with the merge until I get a DOI for the paper describing the revTPSS-NL parameters.

**Comment by hokru** [comment]
> Can you try running wB97M-D3BJ gradients just to double check they work?

They execute fine and deviations between analytical and numerical gradient are below 1e-4, similar to what BP86-D3BJ gives.

**Comment by dgasmith** [comment]
Great thanks, can you add a small test for that in one of the current gradient tests? Easy functionality to accidentally break in the future.

**Comment by hokru** [comment]
I added a short gradient test as requested and the DOI I wanted.
From my side this is ready.

**Comment by PeterKraus** [comment]
https://github.com/psi4/psi4/blob/c33fc0aa50ea52b8a4f8efaba170285b59626dd1/psi4/driver/procrouting/dft/dft_builder.py#L71-L76

Perhaps the new `"nlc"` item could be added to the docs here. Otherwise, excellent work - I'm surprised how non-invasive the changes are!

File: psi4/driver/procrouting/dft/dft_builder.py [hunk 1]
```diff
@@ -364,6 +364,10 @@ def build_superfunctional_from_dictionary(func_dictionary, npoints, deriv, restr
         d_params = func_dictionary["dispersion"]
         if "citation" not in d_params:
             d_params["citation"] = False
+        if "nlc" in d_params:
+            sup.set_vv10_b(-1.0) 
+            sup.set_do_vv10(d_params["nlc"])
+            # sup.set_vv10_b(-1.0) # not here, this re-activates vv10 again.
```
[the remark this answers is not in the document]
Comment (hokru) [reply psi4/driver/procrouting/dft/dft_builder.py]: yes, I will remove it

File: tests/dft-grad-meta/input.dat [hunk 2]
```diff
@@ -46,3 +46,19 @@ analytic = gradient('TPSS', dertype=1)
...
+O         -1.65542       -0.12330        0.00000
+O          1.24621        0.10269        0.00000
+H         -0.70409        0.03193        0.00000
+H         -2.03867        0.75372        0.00000
+H          1.57599       -0.38252       -0.75856
+H          1.57599       -0.38252        0.75856
+}
+
+
+analytic = gradient('B97M-D3BJ', dertype=1)
```
[the remark this answers is not in the document]
Comment (hokru) [reply tests/dft-grad-meta/input.dat]: I thought i had removed this test (in exchange for the faster `dft-grad-lr3` test). The `b97m` has noticeable numerical noise, the `wb97m/x` have no such issue

The aliasing does not seem to work: `Derivative method 'name' b97m-d3(bj) and derivative level 'dertype' 1 are not available.`

[the remark this answers is not in the document]
Comment (hokru) [reply tests/dft-grad-meta/input.dat]: I added manual aliasing.  No aliasing for manually defined `-D3` dictionary functionals (same e.g. for `dsd-pbepbe-d3bj`)

**Pull request merged**